components/elm/src/main/pftvarcon.F90

```diff
@@ -410,6 +434,26 @@ subroutine pftconrd
     expected_pftnames(49) = 'willow                             '
     expected_pftnames(50) = 'irrigated_willow                   '
 
+    ! read actual 'npft' from parameter file
+    if (masterproc) then
+       write(iulog,*) 'Attempting to read PFT physiological data .....'
+    end if
+    call getfil (paramfile, locfn, 0)
+    call ncd_pio_openfile (ncid, trim(locfn), 0)
+    call ncd_inqdid(ncid,'pft',dimid)
+    call ncd_inqdlen(ncid,dimid,npft)
+
+    ! now 'mxpft' in 'elm_varpar' updated by npft here
+    mxpft = npft - 1
+    mxpft_nc = npft - 1     ! when .not.use_crop, so here temporarily set and may be changed after read-through PFT-physiology
+
+    ! In parameter file, 'npft' (mxpft) can be greater than 'maxpatch_pft' in surfdata by namelist setting,
+    ! but cannot less (i.e. pft numbers in 'surfdata'<= that in 'parameters') if non-crop version of model.
```

Comment:
```suggestion
    ! but cannot be less (i.e. there cannot be more pfts in the surface file than there are set of pft parameters) if running the non-crop version of model.
```
Is there a separate crop parameters file? We should ask the crop people about this